lake/passes/passes.py

```diff
@@ -67,25 +63,25 @@ def __init__(self, use_sram_stub, ports):
 
             def visit(self, node):
                 if isinstance(node, Port):
-                    if not(len(node.get_attributes()) <= 0 or self.use_sram_stub):
-                        for i in range(len(node.get_attributes())):
-                            if (isinstance(node.get_attributes()[i].get(), SRAMPortAttr)):
-                                if node.name == "sram_addr":
-                                    node.name = self.sram_macro_info.addr_port_name
-                                elif node.name == "sram_cen":
-                                    node.name = self.sram_macro_info.ce_port_name
-                                elif node.name == "sram_clk":
-                                    node.name = self.sram_macro_info.clk_port_name
-                                elif node.name == "sram_data_in":
-                                    node.name = self.sram_macro_info.data_in_port_name
-                                elif node.name == "sram_data_out":
-                                    node.name = self.sram_macro_info.data_out_port_name
-                                elif node.name == "sram_wen":
-                                    node.name = self.sram_macro_info.wen_port_name
-                                elif node.name == "sram_wtsel":
-                                    node.name = self.sram_macro_info.wtsel_port_name
-                                elif node.name == "sram_rtsel":
-                                    node.name = self.sram_macro_info.rtsel_port_name
+                    attrs = node.find_attribute(lambda a: isinstance(a, SRAMPortAttr))
+                    if len(attrs) != 1:
+                        return
+                    if node.name == "sram_addr":
```

Comment:
Before this set of ifs, there should be a if not self.use_sram_stub check here for the port name replacement to replace only when we are not using Lake's sram_stub. Could add to line 67?